query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java

```diff
@@ -398,15 +394,19 @@ private void performSearchWork(Work work, TransactionContext transactionContext)
    }
 
    private Object extractValue(Object storedValue) {
-      return valueDataConversion.extractIndexable(storedValue);
+      return this.cache.getValueDataConversion().extractIndexable(storedValue);
    }
 
    private Object extractKey(Object storedKey) {
-      return keyDataConversion.extractIndexable(storedKey);
+      if(storedKey == null) return null;
+      DataConversion valueDataConversion = cache.getValueDataConversion();
+      boolean protoWrapper = valueDataConversion.getWrapper().isFilterable();
+      if(protoWrapper) return cache.getKeyDataConversion().getWrapper().unwrap(storedKey);
```

Comment:
Missing whitespace (also on line 401)

Also, the javadoc of `Wrapper.isFilterable()` says `true if the wrapped format is suitable to be indexed or filtered, thus avoiding extra unwrapping`, but here you're doing the unwrapping in the `true` case. Maybe the javadoc is referring to something else?

Reply:
I am not unwrapping the ```value``` (that is filterable), but the ```key``` in this method. Let me see if I can simplify this, I agree it's convoluted